sound/soc/sof/ipc4-topology.c

```diff
@@ -1451,6 +1452,21 @@ static int sof_ipc4_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget
 
 		msg = &pipeline->msg;
 		msg->primary |= pipeline->mem_usage;
+
+		if (ipc4_data->max_num_pipelines <= 0) {
+			dev_err(sdev->dev, "Invalid max_num_pipelines %d\n",
+				ipc4_data->max_num_pipelines);
+			return ipc4_data->max_num_pipelines;
```

Comment:
If `ipc4_data->max_num_pipelines` is 0, then you return here with success?

Reply:
> If `ipc4_data->max_num_pipelines` is 0, then you return here with success?

Moved the test to where we get `ipc4_data->max_num_pipelines` and return -EINVAL in error.